File: ABC/abc235/src/c.rs
use proconio::input;
use std::collections::HashMap;

fn main() {
    input! {
        n: usize,
        q: usize,
        a: [usize; n],
        xk: [(usize, usize); q],
    }

    let mut hashmap = HashMap::new();
    for i in 0..n {
        hashmap.entry(a[i]).or_insert(vec![]).push(i + 1);
    }

    for &(x, k) in xk.iter() {
        if hashmap.contains_key(&x) && hashmap[&x].len() >= k {
            println!("{}", hashmap[&x][k - 1]);
        } else {
            println!("-1");
        }
    }
}

File: abc/abc235/c.py
from collections import defaultdict

n, q = map(int, input().split())
a = list(map(int, input().split()))

hashmap = defaultdict(list)
for i, e in enumerate(a):
    hashmap[e].append(i + 1)

for _ in range(q):
    x, y = map(int, input().split())
    if y - 1 < len(hashmap[x]):
        print(hashmap[x][y - 1])
    else:
        print(-1)
